liblangutil/Scanner.cpp

```diff
@@ -1023,6 +1036,8 @@ std::tuple<Token, unsigned, unsigned> Scanner::scanIdentifierOrKeyword()
 	auto const token = TokenTraits::fromIdentifierOrKeyword(m_tokens[NextNext].literal);
 	switch (m_kind)
 	{
+	case ScannerKind::SourceLocationComment:
+		[[fallthrough]];
 	case ScannerKind::Solidity:
```

Comment:
I think `[[fallthrough]]` is only necessary the case has some code in it.

```suggestion
	case ScannerKind::SourceLocationComment:
	case ScannerKind::Solidity:
```

Reply:
But also, might make more sense to have no keywords in this mode rather than just randomly choose the same keywords as in `Solidity` mode :) This would probably solve the problem with `hex`/`unicode` string literals as those would no longer be recognized.

Reply:
`[[fallthrough]]` is never _really_ necessary I guess; no keywords absolutely makes sense to me, I'm changing that :)

Reply:
But I suppose it would generate a warning at least? If so, then for us it is actually necessary, because we treat warnings as errors :)

But yeah, it's of course not wrong to have it, I pointed it out mostly because we never do it in such cases. And I think that's better because it makes the obvious case less verbose.